driver/controller.go

```diff
@@ -225,6 +276,82 @@ func (s controllerService) ValidateVolumeCapabilities(ctx context.Context, req *
 	}, nil
 }
 
+// CreateSnapshot creates a logical volume snapshot.
+func (s controllerService) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
+	// Since the kubernetes snapshots are Read-Only, we set accessType as 'ro' to activate thin-snapshots as read-only volumes
+	accessType := "ro"
+	// Set snaptype as 'thin' to activate thin-snapshots.
+	// TODO: When adding support for thick-snapshots, set this option as configurable.
+	snapType := "thin"
+
+	ctrlLogger.Info("CreateSnapshot called",
+		"name", req.GetName(),
+		"source_volume_id", req.GetSourceVolumeId(),
+		"parameters", req.GetParameters(),
+		"num_secrets", len(req.GetSecrets()))
+
+	if req.GetSourceVolumeId() == "" {
```

Comment:
type VolumeSnapshotSpec struct {
	// source specifies where a snapshot will be created from.
	// This field is immutable after creation.
	// Required.
	Source VolumeSnapshotSource `json:"source" protobuf:"bytes,1,opt,name=source"`

Is this check required seeing that source is a required field?

Reply:
True! Although it is a required parameter, added this as an extra validation to verify the request.